src/components/shared/device-list.tsx

```diff
@@ -19,44 +10,41 @@ interface DeviceListProps extends
   devices: DeviceWithData[];
 }
 
-interface DeviceListState {
-  deviceDataSource: ListViewDataSource;
-}
+interface DeviceListState {}
 
 export class DeviceList extends React.Component<DeviceListProps, DeviceListState> {
-  connectedImage = require('../../../images/device_on.png');
-  disconnectedImage = require('../../../images/device_off.png');
-
   render() {
     return (
-      <Content>
-        <List
-          enableEmptySections
-          dataArray={this.props.devices}
-          renderRow={(deviceWithData: DeviceWithData) => {
-            const curMessage = deviceWithData.mqttData['_updates/fields'];
-            const device = deviceWithData.device;
-            const curData = curMessage && curMessage.message ? curMessage.message.parsedPayload : null;
-            const connected = curData ? curData.state.connected : false;
-
-            return (
-              <ListItem>
-                <Image
-                  style={Styles.deviceConnectedImage}
-                  source={connected ? this.connectedImage : this.disconnectedImage}
-                />
-
-                <Text style={Styles.listItemText} onPress={() => {
-                  this.props.onPress(deviceWithData);
-                }}>
-                  {device.name || device.serialNumber || '(no name)'} {curData ? curData.state.firmwareVersion : ''}
-                </Text>
-              </ListItem>
-            );
-            }
+      <List
+        enableEmptySections
+        dataArray={this.props.devices}
+        renderRow={(deviceWithData: DeviceWithData) => {
+          const curMessage = deviceWithData.mqttData['_updates/fields'];
```

Comment:
My gut tells me the connected icon should be a seperate component. actually @PCaponetti did we want to start adding the react-native version of the reflections components we were creating in web riot?

Reply:
@harrismendell, I think this 'icon' isn't intended to be only for the connected status but to display different kind of information, for example the device temperature and not sure if it's worth creating a separate component since it's going to be used only in this view. We could create a rendering function that receives the information to be displayed.

Reply:
thats fair, lets just leave it as is for now